pkg/dependencies/factory.go

```diff
@@ -1087,3 +1092,11 @@ func (f *Factory) WithVSphereDefaulter() *Factory {
 
 	return f
 }
+
+func getProxyConfiguration(clusterSpec *cluster.Spec) (string, string, []string) {
```

Comment:
I would name these returns
```suggestion
func getProxyConfiguration(clusterSpec *cluster.Spec) (httpProxy, httpsProxy string, noProxy []string) {
```